Sources/protoc-gen-swift/MessageGenerator.swift

```diff
@@ -130,18 +113,18 @@ class MessageGenerator {
     }
   }
 
-  func generateMainStruct(printer p: inout CodePrinter, file: FileGenerator, parent: MessageGenerator?) {
+  func generateMainStruct(printer p: inout CodePrinter, parent: MessageGenerator?) {
     p.print(
         "\n",
         descriptor.protoSourceComments(),
         "\(visibility)struct \(swiftRelativeName): \(swiftMessageConformance) {\n")
     p.indent()
     if let parent = parent {
-        p.print("\(visibility)static let protoMessageName: String = \(parent.swiftFullName).protoMessageName + \".\(protoMessageName)\"\n")
-    } else if !protoPackageName.isEmpty {
-        p.print("\(visibility)static let protoMessageName: String = _protobuf_package + \".\(protoMessageName)\"\n")
+        p.print("\(visibility)static let protoMessageName: String = \(parent.swiftFullName).protoMessageName + \".\(descriptor.name)\"\n")
+    } else if !descriptor.file.package.isEmpty {
+        p.print("\(visibility)static let protoMessageName: String = _protobuf_package + \".\(descriptor.name)\"\n")
     } else {
-        p.print("\(visibility)static let protoMessageName: String = \"\(protoMessageName)\"\n")
+        p.print("\(visibility)static let protoMessageName: String = \"\(descriptor.name)\"\n")
     }
 
     let usesHeadStorage = storage != nil
```

Comment:
Yea, I spotted this today also. It will get cleaned up in the larger changes to field generation.